[release-0.5] backport CI fixes

vtjnash opened the pull request:
to try making travis happier

vtjnash committed:
try harder to print info when deps target fails on travis

also intersperse stderr info into the log
and reduce the update interval of bar

(cherry picked from commit 1104893c6145489509294d89e0722cf77eaf789d, PR #17859)

vtjnash committed:
ulimit the max open pipes during testing

fix #19132

(cherry picked from commit 87c559c8bd4585903dff839c24ef5b7a1c78d4ec, PR #19307)

vtjnash committed:
update travis xcode version [av skip]

SIP prevents the DYLD variables from being passed to make via the environment
so pass it via the arguments instead

(cherry picked from commit 156edb485fe4d4b804076be0258223884f03195d, PR #19475)

tkelman committed:
Make spawn test depend on current ulimit value instead of changing it

this should allow Base.runtests("spawn") to pass on travis

(cherry picked from commit 9895d789ff0b6fe15df1247defcb0db83abe8f17, PR #19621)

tkelman committed:
Add a contingency for ulimit -n returning unlimited

I have yet to be able to cause this to happen anywhere though

(cherry picked from commit e16453267993d60bb679b7b2979c8e1a09ce0aab, PR #19621)

simonbyrne committed:
start updating for 0.25 (#19855)

(cherry picked from commit 62852fd005d72d2bffb4c31507742bf73bc60a0b)

simonbyrne committed:
Update LibGit2 types for 0.5 (#19855)

(cherry picked from commit f3727333c292a9a65c6c394fdbdc2514cbc2281f,
and rewritten to use the old manual style of constructor definitions)

vtjnash committed:
BUILD_OPTS was not getting exported on linux (#19904)

(cherry picked from commit c5f8375249293a2426648d6ec14184f4bf371432, PR #19904)

vtjnash committed:
fix the fd exhaustion test

the test_broken was on the wrong branch
it would swallow the error + stacktrace if the error wasn't the expected one
and it failed to close all the file descriptors if the test accidentally succeeded

(cherry picked from commit 99d1be7da7d4b39118bc5bb8d746c1656dc6e1bc, PR #20219)

maleadt committed:
Make sure flisp.boot is copied to an existing directory. (#20430)

(cherry picked from commit 86e8039615bb8219a71991e130be5f632d447434)

maleadt commented:
Now that you're backporting CI fixes to 0.5, maybe include #20430? Example [CI failure](https://ci.maleadt.net/buildbot/julia/builders/Julia%200.5%20%28x86-64%29/builds/12).

Rebased patch:
```patch
From 7ff704c8feb084fad062f3e9a014dfc1c7d85c95 Mon Sep 17 00:00:00 2001
From: Tim Besard <tim.besard@gmail.com>
Date: Tue, 7 Feb 2017 13:01:27 +0100
Subject: [PATCH] Make sure flisp.boot is copied to an existing directory.

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index d7ec864f27..9aae087541 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -151,7 +151,7 @@ $(BUILDDIR)/julia_flisp.boot.inc: $(BUILDDIR)/julia_flisp.boot $(FLISP_EXECUTABL
 
 $(BUILDDIR)/julia_flisp.boot: $(addprefix $(SRCDIR)/,jlfrontend.scm \
                julia-parser.scm julia-syntax.scm match.scm utils.scm ast.scm macroexpand.scm mk_julia_flisp_boot.scm) \
-               $(FLISP_EXECUTABLE_release)
+               $(FLISP_EXECUTABLE_release) | $(BUILDDIR)
        @$(call PRINT_FLISP, $(call spawn,$(FLISP_EXECUTABLE_release)) \
                $(call cygpath_w,$(SRCDIR)/mk_julia_flisp_boot.scm) $(call cygpath_w,$(dir $<)) $(notdir $<) $(call cygpath_w,$@))
 
-- 
2.11.1
```

tkelman reviewed (changes requested):
We should not be changing the libgit2 version on release-0.5

tkelman commented on base/libgit2/types.jl:
```diff
@@ -174,7 +174,50 @@ function RemoteCallbacks(credentials::Ptr{Void}, payload::Ref{Nullable{AbstractC
 end
 
-if LibGit2.version() >= v"0.24.0"
+immutable ProxyOptions
```
should put the definition of this type under the `if LibGit2.version() >= v"0.25.0"` as well, it shouldn't be added to the module unless you're building in a non-default configuration and absolutely need it

vtjnash replied:
It's not an API issue. I'm just attempting to roughly match whatever we have on master, not actually redesign that commit.

tkelman replied:
release-0.5 shouldn't match what's on master, it should remain close to 0.5.0. just change msys_build.sh to download 0.5-latest and you won't need to touch libgit2 at all

vtjnash replied:
I don't think semver cares about internal names. This is non-breaking at the API level to keep this slightly closer to master.

tkelman replied:
I don't want to be defining an unused type in the default build configuration.

StefanKarpinski commented:
Can we make some progress on this, pretty please?

tkelman committed:
Use 0.5-latest binary on appveyor

and adjust branch whitelist regex to only count branches that start with release-*

tkelman committed:
Only define ProxyOptions type when using libgit2 >= 0.25

tkelman committed:
also fix branch whitelist regex on travis
